### PR TITLE
fix #2639: getOwningProcessId sometimes return -1 on 64x linux

### LINKED ISSUE #2639: LinuxInternetProtocolStats IPConnection invoke getOwningProcessId always return -1
ProcessStat.querySocketToPidMap return the map key use integer ,but  socketId sometimes use the big int like 2247483647  in linux. so querySocketToPidMap function return Map<Long,Integer> to fix thr bug

### PR DESCRIPTION
**Pull request by yourancc**
fix the issue(#2639 ) where the owning process ID sometimes returns -1 caused by it.

### CODE REVIEW

File: oshi-core/src/main/java/oshi/software/os/linux/LinuxInternetProtocolStats.java [hunk 1]
```diff
@@ -54,15 +54,15 @@ public UdpStats getUDPv6Stats() {
...
+        Map<Long, Integer> pidMap = ProcessStat.querySocketToPidMap();
         conns.addAll(queryConnections("tcp", 4, pidMap));
         conns.addAll(queryConnections("tcp", 6, pidMap));
         conns.addAll(queryConnections("udp", 4, pidMap));
         conns.addAll(queryConnections("udp", 6, pidMap));
         return conns;
     }
 
-    private static List<IPConnection> queryConnections(String protocol, int ipver, Map<Integer, Integer> pidMap) {
+    private static List<IPConnection> queryConnections(String protocol, int ipver, Map<Long, Integer> pidMap) {
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/software/os/linux/LinuxInternetProtocolStats.java:65]: You need to parse the `inode` a few lines down to a long to match this key.

**Review by dbwiddis — COMMENTED** [review]
One fix needed. Also please submit a change log entry!

You can just push a new commit to your branch and it will update this PR.

**Comment by yourancc** [comment]
Thank you for your feedback. I have made the necessary changes to the code.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@
   [#2625](https://github.com/oshi/oshi/pull/2625): Add part number to Physical Memory - [@BartekDziurowicz](https://github.com/BartekDziurowicz), [@dbwiddis](https://github.com/dbwiddis).
 
 ##### Bug fixes / Improvements
+* [#2645](https://github.com/oshi/oshi/pull/2645): fix getOwningProcessId sometimes return -1 on 64x linux - [@yourancc](https://github.com/yourancc).
```
Comment (dbwiddis) [inline CHANGELOG.md:10]: This needs to be under the new 6.6.1 section

**Review by dbwiddis — COMMENTED** [review]
Almost there!

**Review by dbwiddis — APPROVED** [review]
Thanks for your contribution to open source!

**Comment by yourancc** [comment]
thank you for your patient guidance

**Pull request merged**